luci-app-passwall/Makefile

```
@@ -139,38 +146,19 @@ define Package/$(PKG_NAME)/conffiles
 /usr/share/passwall/rules/block_ip
 endef
 
-define Package/$(PKG_NAME)/install
-	$(INSTALL_DIR) $(1)/etc/config
-	$(INSTALL_CONF) ./root/etc/config/passwall $(1)/etc/config/passwall
-	$(INSTALL_CONF) ./root/etc/config/passwall_server $(1)/etc/config/passwall_server
-	
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./root/etc/init.d/passwall $(1)/etc/init.d/passwall
-	$(INSTALL_BIN) ./root/etc/init.d/passwall_server $(1)/etc/init.d/passwall_server
-	
-	$(INSTALL_DIR) $(1)/etc/uci-defaults
-	$(INSTALL_CONF) ./root/etc/uci-defaults/* $(1)/etc/uci-defaults
-	
-	$(INSTALL_DIR) $(1)/usr/share/passwall
-	cp -pR ./root/usr/share/passwall/* $(1)/usr/share/passwall
-	$(INSTALL_CONF) ./root/etc/config/passwall $(1)/usr/share/passwall/config.default
-	
-	$(INSTALL_DIR) $(1)/usr/share/rpcd/acl.d
-	cp -pR ./root/usr/share/rpcd/acl.d/* $(1)/usr/share/rpcd/acl.d
-	
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci
-	cp -pR ./luasrc/* $(1)/usr/lib/lua/luci/
-	
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/i18n
-	po2lmo ./po/zh-cn/passwall.po $(1)/usr/lib/lua/luci/i18n/passwall.zh-cn.lmo
-endef
-
-define Package/$(PKG_NAME)/postinst
+define Package/luci-app-passwall/postinst
 #!/bin/sh
-chmod a+x $${IPKG_INSTROOT}/usr/share/passwall/* >/dev/null 2>&1
-chmod a+x $${IPKG_INSTROOT}/usr/lib/lua/luci/model/cbi/passwall/server/api/app.lua >/dev/null 2>&1
-touch $${IPKG_INSTROOT}/etc/config/passwall_show >/dev/null 2>&1
+
+if [ -z "$${IPKG_INSTROOT}" ]; then
+	(. /etc/uci-defaults/luci-passwall && rm -f /etc/uci-defaults/luci-passwall)
+	touch $${IPKG_INSTROOT}/etc/config/passwall_show >/dev/null 2>&1
+	rm -f /tmp/luci-indexcache
+	rm -rf /tmp/luci-modulecache/
+	killall -HUP rpcd 2>/dev/null
+fi
 exit 0
 endef
 
-$(eval $(call BuildPackage,$(PKG_NAME)))
+include ../../luci.mk
```

```suggestion
include $(TOPDIR)/feeds/luci/luci.mk
```
This one is out-of-tree, so you must specify the absolute path to .mk file.

---

Thank you for your correction